### PR TITLE
Switch distribution from Module::Build to ExtUtils::MakeMaker

### PR DESCRIPTION
**Pull request by karenetheridge**
as discussed, a PR!

The main commit switches the tooling from Module::Build to ExtUtils::MakeMaker, and adjusts some related files.  Note that the release process I'd recommend with this is `perl Makefile.PL; make; make test; make manifest; make dist; cpan-upload Text-Glob-0.10.tar.gz`

I included a few other commits that add some metadata and auto-generate a README (I saw you had a commit recently about this file; I wasn't sure if you dropped it because it was too much bother or if you didn't want it) -- I won't be offended if you omit any of these commits.

### CODE REVIEW

File: lib/Text/Glob.pm [hunk 1]
```diff
@@ -3,7 +3,7 @@ use strict;
 use Exporter;
 use vars qw/$VERSION @ISA @EXPORT_OK
             $strict_leading_dot $strict_wildcard_slash/;
-$VERSION = '0.09';
+$VERSION = '0.10';
```
Comment (richardc) [inline lib/Text/Glob.pm:6]: Please don't touch this, I'll do it ad/when I cut a release.

File: Makefile.PL [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        resources => {
+            repository => {
+                url => 'https://github.com/richardc/perl-text-glob.git',
+                web => 'https://github.com/richardc/perl-text-glob',
+                type => 'git',
+            },
+            bugtracker => {
+                mailto => 'bug-Text-Glob@rt.cpan.org',
+                web => 'https://rt.cpan.org/Public/Dist/Display.html?Name=Text-Glob',
+            },
```
Comment (richardc) [inline Makefile.PL:27]: This is not how I track bugs.   Emails to the author please.

File: Makefile.PL [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    LICENSE             => 'perl_5',
+
+    META_MERGE => {
+        'meta-spec' => { version => 2 },
+        dynamic_config => 0,
+        resources => {
+            repository => {
+                url => 'https://github.com/richardc/perl-text-glob.git',
+                web => 'https://github.com/richardc/perl-text-glob',
+                type => 'git',
```
Comment (richardc) [inline Makefile.PL:22]: If I move where I keep source I have to change this?  No thanks.

File: Makefile.PL [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            },
+            bugtracker => {
+                mailto => 'bug-Text-Glob@rt.cpan.org',
+                web => 'https://rt.cpan.org/Public/Dist/Display.html?Name=Text-Glob',
+            },
+        },
+        x_authority => 'cpan:RCLAMP',
+        x_contributors => [ # manually added, from git shortlog -e -s -n
```
Comment (richardc) [inline Makefile.PL:30]: I don't want more manual steps.

File: Makefile.PL (hunk 2 above)
Comment (karenetheridge) [inline Makefile.PL:27]: Good to know. RT is the default if it's not set, so if you prefer something else, it's best to set it.

File: Makefile.PL (hunk 3 above)
Comment (karenetheridge) [inline Makefile.PL:22]: It's not required, but it's awfully helpful to be able to find the repository for submitting patches -- metacpan.org provides a link using this data.

File: README [hunk 5]
```diff
@@ -0,0 +1,78 @@
+NAME
```
Comment (richardc) [inline README:1]: Do not add auto-generated files to the repository.

File: Makefile.PL [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+delete $WriteMakefileArgs{MIN_PERL_VERSION}
+    if $eumm_version < 6.48;
+
+delete @WriteMakefileArgs{qw(META_ADD META_MERGE)}
+    if $eumm_version < 6.46;
+
+delete $WriteMakefileArgs{LICENSE}
+    if $eumm_version < 6.31;
+
```
Comment (richardc) [inline Makefile.PL:99]: Why all this stuff gated on version?  Do earlier version of MakeMaker reject these parameters?  Will later versions require further updates?  Why are we replacing 12 lines of declarative configuration with a 105 line monster?

**Comment by richardc** [comment]
Please revisit the commit message of the first commit so it fits the general 50 character summar/subject line, then some discussion about what it does.  If you can explain what 'being high in the CPAN river' means, and why that's important it would help, as that sounds like some insider reference I don't get.

**Comment by karenetheridge** [comment]
closed, in favour of #4.

**Pull request closed without merging**